src/cuda/image_function_cuda.cu

```diff
@@ -239,6 +239,59 @@ namespace
         }
     }
 
+    __global__ void rotateCuda( const uint8_t * in, uint32_t rowSizeIn, uint32_t inHeight, float inXStart, float inYStart, 
+                                uint8_t * out, uint32_t rowSizeOut, uint32_t outHeight, 
+                                float cosAngle, float sinAngle)
+    {
+        uint32_t outX = blockDim.x * blockIdx.x + threadIdx.x,
+                 outY = blockDim.y * blockIdx.y + threadIdx.y;
+       
+        // If this thread isn't for a valid pixel in the output image, then do nothing. 
+
+        if ( outX >= rowSizeOut || outY >= outHeight ) 
+            return;
+
+        // Both input coordinates are shifted using the cosAngle, sinAngle, outX, and outY. The shift
+        // comes from inverse rotating the horizontal and vertical iterations over the output.
+
+        // Note that inverse rotation of horizontal motion is ( cos(angle), -sin(angle)),
+        // and the inverse rotation of vertical motion is ( sin(angle), cos(angle) ).
+
+        float exactInX = inXStart + cosAngle * outX + sinAngle * outY,
```

Comment:
Please use `const` word wherever it is possible in variable initialisation :)

Reply:
I used `const` where appropriate. I also removed the pointer variables `inMem` and `outMem` from `rotateCuda()` as they are unnecessary. We can just use `in` and `out` directly.